### ISSUE TITLE
Removing operations outside the docker image

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-smithy#1020]
There have been a few changes that have accumulated in the scripts outside the Docker image as part of the compiler transition. While it seemed reasonable to pick the fastest path forward for the compiler transition, that phase has past. The result of these changes like [this one]( https://github.com/conda-forge/conda-smithy/blob/eb435a5b45568247536b3956abe9cea8ebcb60a5/conda_smithy/templates/run_docker_build.sh.tmpl#L35 ) has been making it harder to cleanly build and test recipes with the Docker image locally. It would be great to adjust things with the Docker image build to avoid reliance on the system configuration outside that build to make testing locally easier.

**Comment by isuruf** [comment]
For the shyaml file, 
DOCKER_IMAGE=$(awk '/docker_image:/{getline; print}' "${FEEDSTOCK_ROOT}/.ci_support/${CONFIG}.yaml" | cut -c 3-)
```
if [[ "${DOCKER_IMAGE}" == "" ]]; then
    DOCKER_IMAGE="{{ docker.image }}"
fi
```
should work

**Comment by jakirkham** [comment]
Another option might be to move the Docker image information back into `conda-forge.yml`.

**Comment by isuruf** [comment]
There are different docker images now for different architectures

**Comment by mariusvniekerk** [comment]
I get *very* nervous about parsing yaml/json with non-json / yaml parsers.  YAML itself is enormously complicated and the chance of something breaking subtly is very high.  
If we want to have some small static thing then we should rather build a statically linked binary and retrieve it, if we wish to avoid installing this.

The other approach is to have a fallback mode that just uses `pyyaml` which you're likely to have if you invoke conda yourself.

**Comment by jakirkham** [comment]
> There are different docker images now for different architectures

My knowledge of how this is being handled currently is weak. More info would be helpful.

> I get _very_ nervous about parsing yaml/json with non-json / yaml parsers. YAML itself is enormously complicated and the chance of something breaking subtly is very high.

That seems like a fair concern to me. Would be happy to hear other ways to solve this.

> The other approach is to have a fallback mode that just uses `pyyaml` which you're likely to have if you invoke conda yourself.

That might work. In this case we can probably run the Docker image to do the parsing. It's a bit of a heavyweight fix though.

**Comment by isuruf** [comment]
the `.ci_support/*.yaml` files use a small subset of YAML. I don't think it's a huge issue.
Other option is to have a `DOCKER_IMAGE` variable set in the CI files and not use shyaml when that env variable is set.

**Comment by mariusvniekerk** [comment]
For the binary side we could use https://github.com/mikefarah/yq which is a standalone golang based binary

**Comment by isuruf** [comment]
You still have to download for different architectures after detecting the native platform which is messy. I thin setting a `DOCKER_IMAGE` variable is the easiest approach here.

**Comment by mariusvniekerk** [comment]
yeah, probably the docker_image with some fallback to the defaults comp7 image, that way people who just run the build locally could still succeed.   Alternatively, we could explicitly add a local test executable that inspects the variant etc, and makes local running simpler

**Comment by scopatz** [comment]
FWIW, I have never really gotten building locally with a docker container working, and have a tried a few times over the years.  It might be nice if conda-smithy produced a run locally script that you could use.

**Comment by isuruf** [comment]
I've always been able to run the script until shyaml came in.
